src/hipscat_import/file_readers.py

```diff
@@ -0,0 +1,37 @@
+"""File reading generators for common file types."""
+
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as pq
+from astropy.table import Table
+
+
+def get_file_reader(file_format):
+    """Get a generator file reader for common file types"""
+    if "csv" in file_format:
+        return csv_reader
+    if file_format == "fits":
+        return fits_reader
+    if file_format == "parquet":
+        return parquet_reader
```

Comment:
I think this could be refactored further. We could have a [factory method](https://www.geeksforgeeks.org/factory-method-python-design-patterns/) to generate the correct file_reader from the file_format string.

Reply:
My reading of the factory as linked is that you'd instantiate all the types and only call their methods when the appropriate arguments come up. In this case, we'd only want to instantiate one kind of reader for the whole execution of the pipeline, so the setup of the factory would be unnecessary overhead. Happy to talk further (or punt on more changes for now).

Reply:
I think that example instantiates all the types because that's what they want, but the factory itself just instantiates a single type. And it does so by looking up a string into a dictionary instead of if statements, which is neater and easier to add more formats to, including that you can do it programmatically. (If you really wanted to get fancy, you could use a decorator to add new file formats without having to change the factory class at all)

Reply:
But yeah, it's not necessary right now if we just want to get this merged